pkgs/build-support/make-binary-wrapper/src/lib/libwrapper/wrapper.nim

```diff
@@ -0,0 +1,102 @@
+# This is the code of the wrapper that is generated.
+
+import json
+import os
+import posix
+import sequtils
+import strutils
+
+# Wrapper type as used by the wrapper-generation code as well in the actual wrapper.
+
+type
+    SetVar* = object
+        variable*: string
+        value*: string
```

Comment:
I'd like it to be possible to `SetVar` multiple values from an array in `wrapper.json`.

Reply:
The way it is now you can have
```json
    "environment": {
        "set": [
            {
                "variable": "HELLO",
                "value": "foo"
            },
            {
                "variable": "HELLO",
                "value": "bar"
            }
        ]
    }
```
Am I correct you are suggesting you want the following?
```json
    "environment": {
        "set": [
            {
                "variable": "HELLO",
                "values": [ 
                    "foo",
                    "bar"
                ]
            }
        ],
```

Reply:
Yes I want it to be an array, and also there should be a `separator` key as well, because some env vars use e.g `;` and not `:`. `:` should be the default though.

Am I correct to understand there's both `SetVar` and `environment.set`?

Reply:
`environment.set` is a sequence of `SetVar` objects where `SetVar` has the fields `variable `and `value`.

I see what you're after. What I did here is basically take the `makeWrapper` arguments as is. If you want `export HELLO="foo:bar"` then in case of `makeWrapper` write `--set HELLO "foo:bar"`, and so you have variable `HELLO` and value `foo:bar`. In that case one does not need to know what separator is used.